portality/bll/services/todo.py

```diff
@@ -138,6 +153,116 @@ def maned_assign_pending(cls, size):
+                    constants.APPLICATION_STATUS_REJECTED,
+                    constants.APPLICATION_STATUS_COMPLETED
+                ])
+            ],
+            sort="last_manual_update",
+            size=size
+        )
+        return constants.TODO_EDITOR_STALLED, stalled, "last_manual_update", False
+
+    @classmethod
+    def editor_follow_up_old(cls, groups, size):
+        follow_up_old = TodoQuery(
+            musts=[
+                TodoQuery.cd_older_than(8),
+                TodoQuery.editor_groups(groups)
+            ],
+            must_nots=[
+                TodoQuery.status([
+                    constants.APPLICATION_STATUS_ACCEPTED,
+                    constants.APPLICATION_STATUS_REJECTED,
+                    constants.APPLICATION_STATUS_COMPLETED
+                ])
+            ],
+            sort="created_date",
+            size=size
+        )
+        return constants.TODO_EDITOR_FOLLOW_UP_OLD, follow_up_old, "created_date", False
+
+    @classmethod
+    def editor_completed(cls, groups, size):
+        completed = TodoQuery(
+            musts=[
+                TodoQuery.status([constants.APPLICATION_STATUS_COMPLETED]),
+                TodoQuery.editor_groups(groups)
+            ],
+            sort="last_manual_update",
+            size=size
+        )
+        return constants.TODO_EDITOR_COMPLETED, completed, "last_manual_update", True
+
+    @classmethod
+    def editor_assign_pending(cls, groups, size):
+        assign_pending = TodoQuery(
+            musts=[
+                TodoQuery.editor_groups(groups),
+                TodoQuery.status([constants.APPLICATION_STATUS_PENDING])
+            ],
+            must_nots=[
+                TodoQuery.exists("admin.editor")
+            ],
+            sort="created_date",
+            size=size
+        )
+        return constants.TODO_EDITOR_ASSIGN_PENDING, assign_pending, "created_date", False
+
+    @classmethod
+    def associate_stalled(cls,  size):
+        sort_field = "last_manual_update"
+        stalled = TodoQuery(
+            musts=[
+                TodoQuery.lmu_older_than(3),
+            ],
+            sort=sort_field,
+            size=size
+        )
+        return constants.TODO_ASSOCIATE_PROGRESS_STALLED, stalled, sort_field, False
+
+    @classmethod
+    def associate_follow_up_old(cls,  size):
+        sort_field = "created_date"
+        follow_up_old = TodoQuery(
+            musts=[
+                TodoQuery.cd_older_than(6),
+            ],
+            must_nots=[
+                TodoQuery.status([
+                    constants.APPLICATION_STATUS_ACCEPTED,
+                    constants.APPLICATION_STATUS_REJECTED,
+                ])
+            ],
+            sort=sort_field,
+            size=size
+        )
+        return constants.TODO_ASSOCIATE_FOLLOW_UP_OLD, follow_up_old, sort_field, False
+
+    @classmethod
+    def associate_start_pending(cls, acc_id, size):
```

Comment:
this rule required condition `As soon as it’s assigned`, 
I'm not sure the query implementation correctly, please take a look.